include/dxguids/dxguids.h

```diff
@@ -55,6 +55,7 @@ template <> constexpr GUID uuidof<InterfaceName>() \
 #endif /* _WIN32 */
 
 // Direct3D
+#ifdef __d3d12_h__
```

Comment:
Yeah, I see where you're going with this. The original intention here was to always include this after D3D12, but I'm remembering that you're trying to get the DXIL header building, which relies on the same IID infrastructure/defines. Given that, yeah I think this is okay - I was tempted to suggest adding a `#warning` if none of these headers were included, but I think you do have a valid scenario here.

Reply:
I think the `#warning` should be in `d3d12.h` and alike files.
That's makes more sense.

Reply:
Before this patch, the compiling error is rahter annoy, too long to debugging.